src/resources/js/admin/tickets-commerce-settings.js

```diff
@@ -250,6 +251,43 @@ tribe.tickets.admin.commerceSettings = {};
 			obj.disconnectButton.on( 'click', obj.handleDisconnectClick );
 		}
 
+		$document.on( 'click', obj.selectors.webHookSetUpLink, function( evt ) {
+			evt.preventDefault();
+			const target = $( evt.target );
+
+			$document.find( '#tec-tickets-webhook-creation-update' ).remove();
+
+			$(
+				`<p id="tec-tickets-webhook-creation-update" class="tooltip description">
+				<span class="dashicons dashicons-update"></span>
+				<span>${ target.attr( 'data-loading-text' ) }</span>
+				</p>`,
+			).insertAfter( target.closest( 'p' ) );
+
+			fetch( target.attr( 'href' ) ).then( function( res ) {
+				return res.json();
+			} ).then( function( res ) {
+				const updateEle = $( '#tec-tickets-webhook-creation-update' );
+				const updateDashIcon = updateEle.find( 'span.dashicons' );
+				const updateText = updateEle.find( 'span:not(.dashicons)' );
+
+				updateText.text( res.data.status );
+
+				updateDashIcon.removeClass( 'dashicons-update' );
+
+				if ( true !== res.success ) {
+					updateDashIcon.addClass( 'dashicons-no' );
+					return;
+				}
+
+				updateDashIcon.addClass( 'dashicons-yes' );
+
+				setTimeout( function() {
+					window.location.reload();
+				}, 700 );
```

Comment:
Is this required?

Reply:
Its not required, but it does solve 2 issues.

1. We don't have to update the field through JS.
2. When the field is being updated from the Frontend, during save JS will attempt to verify the webhook secret again.

With refresh, we have already stored a validated secret during the ajax request, then we just display it validated already.